src/main/java/org/apache/mesos/specification/TaskTypeSpecification.java

```diff
@@ -1,8 +1,16 @@
 package org.apache.mesos.specification;
 
+import org.apache.mesos.Protos;
+
+import java.util.Collection;
+
 /**
- * A TaskTypeSpecification describes the count of a particular TaskSpecification which should be launched.
+ * A TaskTypeSpecification describes a Type of Task and how many should be launched.
  */
-public interface TaskTypeSpecification extends TaskSpecification {
+public interface TaskTypeSpecification {
```

Comment:
Why does the id need to be passed into just the command function? Would name/resources/volumes need it too?

Feels like someone could have any id/type handling inside of their implementation, though that implies the developer owns the creation of TaskTypeSpecification objects.